### DISCUSSION TITLE
Can Khepri store huge amounts of data? Can replace MySQL or store IM chat history data as a sequential database

### DISCUSSION THREAD

**Discussion by leeyisoft** [rabbitmq/khepri#104] [Q&A] [answered]
khepri 能否存储海量数据吗？能否替代MySQL，或者作为时序数据库存储 IM 的聊天历史数据

Can Khepri store huge amounts of data? Can replace MySQL or store IM chat history data as a sequential database

**Comment by dumbbell** [comment, marked as answer]
This is not a scenario we tested so far, but I would say that no, it is **not suited for a huge amount of data currently** for the following reasons:
1. The entire tree structure is kept in memory.
2. The state, which contains that tree structure, is passed to Ra for snapshotting as one structure and is therefore serialized entirely with each snapshot.

**Reply by michaelklishin** [reply]
While this may change eventually, RabbitMQ's needs will be limited to small to moderate data sets in practice.

**Reply by michaelklishin** [reply]
…because we plan on using Khepri only for metadata and not for message stores or streams.
